app/services/update_user_password.rb

```diff
@@ -0,0 +1,47 @@
+class UpdateUserPassword
+  delegate :recovery_code, to: :encryptor
+
+  def initialize(user, user_session, password)
+    @user = user
+    @user_session = user_session
+    @password = password
+  end
+
+  def call
+    form = UpdateUserPasswordForm.new(user)
+    form.submit(password)
+    success = form.valid?
+    process_valid_submission if success
+    FormResponse.new(success: success, errors: form.errors.messages)
```

Comment:
oh ok, it this is an existing pattern I am ok with returning it. 

As background: in my mental model, service classes *do* things but usually do not have a return value that I care about. But this might not be a good mental model :)